src/backend/distributed/planner/distributed_planner.c

```diff
@@ -1413,9 +1423,19 @@ AdjustReadIntermediateResultCost(RangeTblEntry *rangeTableEntry, RelOptInfo *rel
 			rowSizeEstimate += 1;
 		}
 
+
 		/* add the cost of parsing a column */
-		rowCost += get_func_cost(inputFunctionId) * cpu_operator_cost;
+#if PG_VERSION_NUM >= 120000
+		add_function_cost(NULL, inputFunctionId, NULL, &funcCost);
+#else
+		funcCost += get_func_cost(inputFunctionId);
+#endif
 	}
+#if PG_VERSION_NUM >= 120000
+	rowCost += funcCost.per_tuple;
```

Comment:
What I meant is shouldn't we need do add `startup` cost  as below:

```C
diff --git a/src/backend/distributed/planner/distributed_planner.c b/src/backend/distributed/planner/distributed_planner.c
index 312024b1c..c4feffcb5 100644
--- a/src/backend/distributed/planner/distributed_planner.c
+++ b/src/backend/distributed/planner/distributed_planner.c
@@ -1448,7 +1448,13 @@ AdjustReadIntermediateResultCost(RangeTblEntry *rangeTableEntry, RelOptInfo *rel
        /* tell the planner about the cost and row count of the function */
        path = (Path *) linitial(pathList);
        path->rows = rowCountEstimate;
+
+#if PG_VERSION_NUM >= 120000
+       path->total_cost = rowCountEstimate * rowCost + ioCost + funcCost.startup;
+#else
        path->total_cost = rowCountEstimate * rowCost + ioCost;
+#endif
+
 }
```

Reply:
I've also included `relOptInfo->baserestrictcost.startup`